File: sdk-node/examples/features/retrieve.ts
/**
 * Retrieve a single feature by ID.
 *
 * Run:
 *   npx tsx examples/features/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const feature = await client.features.retrieve('feat_replace_with_real_id')

console.log(`feature ${feature.id ?? '?'} [${feature.type ?? '?'}]`)
console.log(`  key   ${feature.key ?? '?'}`)
console.log(`  name  ${feature.name ?? '?'}`)
console.log(`  active ${String(feature.active ?? false)}`)
if (feature.enumValues !== undefined) {
  console.log(`  values ${feature.enumValues.join(', ')}`)
}

File: sdk-python/examples/features/retrieve.py
"""Run with: python examples/features/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        feature = client.features.retrieve("feat_replace_with_real_id")
        print(f"feature {feature.id} [{feature.type}]")
        print(f"  key    {feature.key}")
        print(f"  name   {feature.name}")
        print(f"  active {feature.active}")
        if feature.enum_values is not None:
            print(f"  values {', '.join(feature.enum_values)}")


if __name__ == "__main__":
    main()
